File: algorithms/Implement Stack using Queues/ImplementStackUsingQueues.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Date    : 2016-11-30 21:17:38
# @Author  : luckcul (tyfdream@gmail.com)
# @Version : 2.7.12
from collections import deque
class Stack(object):
    def __init__(self):
        """
        initialize your data structure here.
        """
        self.Queue = deque([])

    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """
        self.Queue.append(x)

    def pop(self):
        """
        :rtype: nothing
        """
        l = len(self.Queue)
        if l == 0:
        	return
        for i in range(l-1):
        	tmp = self.Queue.popleft()
        	self.Queue.append(tmp)
        self.Queue.popleft()


    def top(self):
        """
        :rtype: int
        """
        l = len(self.Queue)
        if l == 0:
        	return None
        tmp = 0
        for i in range(l):
        	tmp = self.Queue.popleft()
        	self.Queue.append(tmp)
        return tmp

    def empty(self):
        """
        :rtype: bool
        """
        return True if len(self.Queue) == 0 else False
        

File: algorithms/Implement Stack using Queues/ImplementStackUsingQueues.cpp
/* ***********************************************
 	Author        : luckcul
 	Mail          : tyfdream@gmail.com
 	Created Time  : 2016-11-30 21:43:03
 	Problem       : problem
************************************************ */
#include <cstdio>
#include <cstring>
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>
#include <cmath>
#include <queue>
#include <map>
#include <set>
using namespace std; 
#define INF 1000000000
//typedef __int64 LL; 

class Stack {
	queue<int> Q;
public:
    // Push element x onto stack.
    void push(int x) {
        Q.push(x);
    }

    // Removes the element on top of the stack.
    void pop() {
        int l = Q.size();
        if(l == 0) return ; 
        for(int i = 0; i < l-1; i++){
        	int tmp = Q.front();
        	Q.pop();
        	Q.push(tmp);
        }
        Q.pop();
    }

    // Get the top element.
    int top() {
        int l = Q.size(), tmp;
        if(l == 0) return NULL;
        for(int i = 0; i < l; i++){
        	tmp = Q.front();
        	Q.pop();
        	Q.push(tmp);
        }
        return tmp;
    }

    // Return whether the stack is empty.
    bool empty() {
        return Q.empty();
    }
};
